client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -275,6 +276,16 @@ class CalypsoifyIframe extends Component< Props & ConnectedProps & ProtectedForm
 			const { closeUrl } = this.props;
 			ports[ 0 ].postMessage( `${ window.location.origin }${ closeUrl }` );
 		}
+
+		// Pipes errors in the iFrame context to the Calypso error handler if it exists:
+		if ( EditorActions.LogError === action ) {
+			const { error } = payload;
+			if ( isArray( error ) && error.length > 3 && window.onerror ) {
+				const errorObject = error[ 4 ];
+				error[ 4 ] = errorObject && JSON.parse( errorObject );
+				window.onerror( ...error );
```

Comment:
Should we just explicitly log here instead of having the parent window call this? I think that'd be a little clearer.

I'm not having a ton of luck tracing actual test errors to kibana, so it'd be good to get a +1 verification on that.

Reply:
eg `import log from 'lib/catch-js-errors/log';`

Reply:
The reason I avoided that is because I wanted to maintain whatever magic TraceKit uses to report the error. Looking at the sourcecode, the log method wouldn't capture the full error: https://github.com/Automattic/wp-calypso/blob/08a39e85303d80e8baf886ea0172c04fcf1dc730/client/lib/catch-js-errors/index.js#L115-L122

(e.g. it would only report the string message if we passed it in, and a stacktrace would probably be the log method and then the iFrame onMessage handler without any of the juicy details from the other side of the iFrame.)

I do agree that it would be clearer to read that way, but I wanted to make sure the uncaught error reporting was triggered in the same way. I'm not sure if there's a way to do that without modifying the library.

> I'm not having a ton of luck tracing actual test errors to kibana

For sure. Errors seem to get sent to `/rest/v1.1/js-error?http_envelope=1` in the public API, so we'd have to track that down

Reply:
Gotcha, maybe leave a comment here to provide context on the trace/logging